src/game/shared/tf/tf_playeranimstate.cpp

```diff
@@ -1461,7 +1461,7 @@ bool CTFPlayerAnimState::HandleJumping( Activity &idealActivity )
 		else if ( ( GetBasePlayer()->GetFlags() & FL_ONGROUND ) == 0 )
 		{
 			// In an air walk.
-			if ( m_pTFPlayer->m_Local.m_flFallVelocity > PLAYER_MAX_SAFE_FALL_SPEED && m_pTFPlayer->m_Shared.CanFallStomp() )
+			if ( m_pTFPlayer->m_Local.m_flFallVelocity > PLAYER_MAX_SAFE_FALL_SPEED && m_pTFPlayer->m_Shared.CanFallStomp() && !( SelectWeightedSequence( TranslateActivity( ACT_MP_FALLING_STOMP ) ) < 0 ) )
```

Comment:
Sorry, why did you write the condition in such a convulted way? Just check the result is greater than or equals to 0.